DevelopGitHub/GitHubExtension/GitHubExtension.LocalizationTool/Translate/Translator.cs

```diff
@@ -0,0 +1,163 @@
+        }
+
+        public static string GetLang(Lang lang)
+        {
+            switch (lang)
+            {
+                case Lang.Uk:
+                    return "uk-UK";
+                case Lang.Us:
+                    return "en-US";
+                case Lang.Ru:
+                    return "ru-RU";
+                default:
+                    throw new Exception("Incorrect Lang in Method GetLang");
+            }
+        }
+
+        public static string GetFileName(Lang lang)
+        {
+            switch (lang)
+            {
+                case Lang.Uk:
+                    return "uk-UK.json";
+                case Lang.Us:
+                    return "en-US.json";
+                case Lang.Ru:
+                    return "ru-RU.json";
+                default:
+                    throw new Exception("Incorrect Lang in Method GetLang");
+            }
+        }
+
+        public async Task WebTranslate(string sourceLanguage, string targetLanguage)
+        {
+            if (sourceLanguage.Equals(targetLanguage))
+            {
+                return;
+            }
+
+            Lang sourceLanguageEnum;
+            Lang targetLanguageEnum;
+            LanguageCheck(ref sourceLanguage, ref targetLanguage, out sourceLanguageEnum, out targetLanguageEnum);
+            StringBuilder textToTranslate = GenerateTextParameter(sourceLanguageEnum);
+            var result = string.Empty;
+            using (var webClient = new WebClient { Encoding = Encoding.UTF8 })
+            {
+                try
+                {
+                    result = await webClient.DownloadStringTaskAsync(
+                                YandexTranslateApiUrl + sourceLanguage + Dash + targetLanguage + textToTranslate);
+                }
+                catch (WebException exception)
+                {
+                    MainWindowViewModel.ShowErrorMessageBox("The waiting time of server response has expired", exception);
+                }
+            }
+            
+            SaveTranslationResult(result, targetLanguageEnum);
+        }
+
+        private static void LanguageCheck(
+            ref string sourceLanguage, 
+            ref string targetLanguage, 
+            out Lang sourceLanguageEnum, 
+            out Lang targetLanguageEnum)
+        {
+            sourceLanguageEnum = GetLang(sourceLanguage);
+            targetLanguageEnum = GetLang(targetLanguage);
+            sourceLanguage = sourceLanguage == UsLang ? EnLang : sourceLanguage;
+            targetLanguage = targetLanguage == UsLang ? EnLang : targetLanguage;
+        }
+
+        private void SaveTranslationResult(string result, Lang language)
+        {
+            if (string.IsNullOrWhiteSpace(result))
+            {
+                return;
+            }
+
+            var jsonResult = JObject.Parse(result);
+            var i = 0;
+            foreach (var value in jsonResult.GetValue(JsonTextParameter).Select(item => item.ToString()))
```

Comment:
Please don't use deferred loading inside foreach because it's evaluated every time when you iterate on elements.